fiftyone/utils/labelbox.py

```diff
@@ -531,87 +462,160 @@ def upload_data(
         task = lb_dataset.create_data_rows(upload_info)
         task.wait_till_done()
 
-    def upload_samples(
-        self,
-        samples,
-        label_schema,
-        media_field="filepath",
-        classes_as_attrs=True,
-        project_name=None,
-        upload_annotations=False,
-        invite_users=[],
-    ):
-        """Parse the given samples and use the label schema to create project,
-        upload data, and upload formatted annotations to Labelbox.
+    def upload_samples(self, samples, backend):
+        """Uploads the given samples to Labelbox according to the given
+        backend's annotation and server configuration.
 
         Args:
             samples: a :class:`fiftyone.core.collections.SampleCollection` to
-                upload to Labelbox
-            label_schema: a dictionary containing the description of label
-                fields, classes and attribute to annotate
-            media_field ("filepath"): string field name containing the paths to
-                media files on disk to upload
-            classes_as_attrs (True): whether to show every class at the top
-                level of the editor (False) or whether to show the label field
-                at the top level and annotate the class as a required
-                attribute (True)
-            project_name (None): the name of the project that will be created,
-                defaults to FiftyOne_<dataset-name>
-            upload_annotations (False): whether to upload annotations to Labelbox.
-                This is considered "Model Assisted Labeling" and is only
-                available for paid Labelbox accounts
-            invite_users ([]): a list of (email, role) tuples specifying the users
-                to invite and their roles in the created project. Options for roles
-                are ["LABELER", "REVIEWER", "TEAM_MANAGER", "ADMIN"]
-
-        Returns: 
-            the ID of the created Labelbox project
+                upload to CVAT
+            backend: a :class:`LabelboxBackend` to use to perform the upload
+
+        Returns:
+            a :class:`LabelboxAnnotationResults`
         """
+        config = backend.config
+        label_schema = config.label_schema
+        media_field = config.media_field
+        project_name = config.project_name
+        members = config.members
+        classes_as_attrs = config._classes_as_attrs
+
         if not classes_as_attrs:
-            # @todo
+            # @todo implement this
             raise NotImplementedError(
-                "Annotating classes at the top level is not yet supported."
+                "Annotating classes at the top level is not yet supported"
             )
+
```

Comment:
The current version raises an error when attempting to upload existing labels. This means that even the updated existing label workflow where you create and upload a new label field with the tagged labels will raise an error. 

I'm not sure if it should be in this PR or a separate one, but we need to add a check/parameter to determine if the user has a paid labelbox account or not. This is what the `upload_annotations` arg was doing previously.

Reply:
It raises an error because uploading existing labels isn't implemented yet :)

I updated the "edit existing labels using free Labelbox" workflow to not use an existing field. Instead it recommends populating a new field which is then merged into the existing field within FiftyOne

Reply:
My issue with `upload_annotations=False` is that I see it purely as a workaround to make the "edit existing labels using free Labelbox" workflow in this work possible with a handful less keystrokes. It's not something I think we should expose in general because it says "hey, you have existing labels but we'll make it easy for you add more (likely duplicate) labels to the field".

I like forcing the user to do a manual workaround if they really want to do this workflow. We put in the work to support uploading existing labels for a "proper" editing experience, so I think the user should see the error to this effect and then either (a) upgrade to paid Labelbox, or (b) switch to CVAT backend to have a proper editing experience

Reply:
Ah I see the issue, I missed the `@todo retrieve label schema for existing 'ground_truth' field` in the existing labels workflow. Without that I assumed you wanted to allow automatic uploading of existing label schemas.

I agree that requiring users to manually provide the label schema and create a new field is the correct way for us to do this. 

Will the `SampleCollection.get_label_schema("ground_truth")` method be added in a separate PR?

Reply:
I'll add it, just need to find the time